Dockerfile

```diff
@@ -356,6 +356,20 @@ RUN --mount=type=cache,target=/.cache/go-build GOOS=linux GOARCH=amd64 go test -
 FROM scratch AS integration-test
 COPY --from=integration-test-build /src/integration.test /integration-test
 
+# The talos target builds talos test binary.
+
+FROM base AS talos-test-build
+ARG SHA
+ARG TAG
+ARG VERSION_PKG="github.com/talos-systems/talos/pkg/version"
+COPY ./hack/test /src/hack/test
+WORKDIR /src/hack/test
+RUN --mount=type=cache,target=/.cache/go-build GOOS=linux GOARCH=amd64 go build -ldflags "-s -w -X ${VERSION_PKG}.Name=Client -X ${VERSION_PKG}.SHA=${SHA} -X ${VERSION_PKG}.Tag=${TAG}" -o talos-test .
+RUN chmod +x ./talos-test
+
+FROM scratch AS talos-test
+COPY --from=talos-test-build /src/hack/test/talos-test /talos-test
+
```

Comment:
I think for these tests (basic integration, e2e, and conformance) we should just use `go run`. The integration test is a bit different since we will publish it.

Reply:
fixed in amended commit.